src/signOpCert.c

```diff
@@ -0,0 +1,198 @@
+		view_memmove(ctx->kesPublicKey, &view, KES_PUBLIC_KEY_LENGTH);
+		TRACE("KES key:");
+		TRACE_BUFFER(ctx->kesPublicKey, KES_PUBLIC_KEY_LENGTH);
+
+		VALIDATE(view_remainingSize(&view) >= 8, ERR_INVALID_DATA);
+		ctx->kesPeriod = parse_u8be(&view);
+		TRACE("KES period:");
+		TRACE_UINT64(ctx->kesPeriod);
+
+		VALIDATE(view_remainingSize(&view) >= 8, ERR_INVALID_DATA);
+		ctx->issueCounter = parse_u8be(&view);
+		TRACE("Issue counter:");
+		TRACE_UINT64(ctx->issueCounter);
+
+		view_skipBytes(&view, bip44_parseFromWire(&ctx->poolColdKeyPathSpec, VIEW_REMAINING_TO_TUPLE_BUF_SIZE(&view)));
+
+		ASSERT(view_remainingSize(&view) == 0);
+	}
+
+	// Check security policy
+	security_policy_t policy = policyForSignOpCert(&ctx->poolColdKeyPathSpec);
+	ENSURE_NOT_DENIED(policy);
+
+	{
+		uint8_t opCertBodyBuffer[OP_CERT_BODY_LENGTH];
+		write_view_t opCertBodyBufferView = make_write_view(opCertBodyBuffer, opCertBodyBuffer + OP_CERT_BODY_LENGTH);
+
+		view_appendData(&opCertBodyBufferView, (const uint8_t*) &ctx->kesPublicKey, SIZEOF(ctx->kesPublicKey));
+		{
+			uint8_t chunk[8];
+			u8be_write(chunk, ctx->issueCounter);
+			view_appendData(&opCertBodyBufferView, chunk, SIZEOF(chunk));
+		}
+		{
+			uint8_t chunk[8];
+			u8be_write(chunk, ctx->kesPeriod);
+			view_appendData(&opCertBodyBufferView, chunk, SIZEOF(chunk));
+		}
+
+		ASSERT(view_processedSize(&opCertBodyBufferView) == OP_CERT_BODY_LENGTH);
+		TRACE_BUFFER(opCertBodyBuffer, SIZEOF(opCertBodyBuffer));
+
+		getOpCertSignature(
+		        &ctx->poolColdKeyPathSpec,
+		        opCertBodyBuffer,
+		        OP_CERT_BODY_LENGTH,
+		        ctx->signature,
+		        SIZEOF(ctx->signature)
+		);
+	}
+	ctx->responseReadyMagic = RESPONSE_READY_MAGIC;
+
+	switch (policy) {
+#	define  CASE(policy, step) case policy: {ctx->ui_step = step; break;}
+		CASE(POLICY_PROMPT_WARN_UNUSUAL,    UI_STEP_WARNING);
+		CASE(POLICY_PROMPT_BEFORE_RESPONSE, UI_STEP_CONFIRM_START);
+		CASE(POLICY_ALLOW_WITHOUT_PROMPT,   UI_STEP_RESPOND);
+#	undef   CASE
+	default:
+		THROW(ERR_NOT_IMPLEMENTED);
+	}
+	signOpCert_ui_runStep();
+}
+
+static void signOpCert_ui_runStep()
+{
+	ui_callback_fn_t* this_fn = signOpCert_ui_runStep;
+
+	UI_STEP_BEGIN(ctx->ui_step, this_fn);
+
+	UI_STEP(UI_STEP_WARNING) {
+		ui_displayPaginatedText(
+		        "Unusual request",
+		        "Proceed with care",
+		        this_fn
+		);
+	}
+	UI_STEP(UI_STEP_CONFIRM_START) {
+		ui_displayPrompt(
+		        "Start new",
+		        "operational certificate?",
+		        this_fn,
+		        respond_with_user_reject
+		);
+	}
+	UI_STEP(UI_STEP_DISPLAY_POOL_COLD_KEY_PATH) {
+		ui_displayPathScreen("Pool cold key path", &ctx->poolColdKeyPathSpec, this_fn);
+	}
+	UI_STEP(UI_STEP_DISPLAY_POOL_ID) {
+		uint8_t poolKeyHash[POOL_KEY_HASH_LENGTH];
+		bip44_pathToKeyHash(&ctx->poolColdKeyPathSpec, poolKeyHash, SIZEOF(poolKeyHash));
+
+		ui_displayBech32Screen(
+		        "Pool ID",
+		        "pool_vk",
```

Comment:
This should be `pool` as it's the pool key hash: https://github.com/cardano-foundation/CIPs/blob/master/CIP-0005/CIP-0005.md#hashes